tests/unit/modules/state_test.py

```diff
@@ -0,0 +1,990 @@
+# -*- coding: utf-8 -*-
+'''
+    :codeauthor: :email:`Rahul Handay <rahulha@saltstack.com>`
+'''
+
+# Import Python libs
+from __future__ import absolute_import
+import os
+
+# Import Salt Testing Libs
+import salt.utils
+from salttesting import TestCase, skipIf
+from salt.exceptions import SaltInvocationError
+from salttesting.mock import (
+    MagicMock,
+    patch,
+    mock_open,
+    NO_MOCK,
+    NO_MOCK_REASON
+)
+
+# Import Salt Libs
+from salt.modules import state
+
+# Globals
+state.__salt__ = {}
+state.__context__ = {}
+state.__opts__ = {}
+state.__pillar__ = {}
+
+
+class MockState(object):
+    '''
+        Mock class
+    '''
+    def __init__(self):
+        pass
+
+    class State(object):
+        '''
+            Mock state class
+        '''
+        flag = None
+
+        def __init__(self, opts, pillar=False):
+            pass
+
+        def verify_data(self, data):
+            '''
+                Mock verify_data method
+            '''
+            data = data
```

Comment:
This statement is technically not necessary since variables that are written into the function signature are scoped into the function body as ordinary variables.